app/views/matches/index.html.slim

```diff
@@ -4,93 +4,11 @@
   a.btn[href="/matches/new"]
     | ADD
 
-- @matches.includes(:player_matches, :players).each do |match|
-  table.match-tbl
-    thead
-      tr
-        - if @date_var != pretty_format_date(match.created_at)
-          - @date_var = pretty_format_date(match.created_at)
-          th[colspan="5"]
-            = @date_var
-
+- @matches.includes(:player_matches, :players).reverse_each do |match|
```

Comment:
more rails magic (can do next time)
`= render @matches.includes(:player_matches, :players).reverse`

then in `_match.html.slim` you can access the variable `match`

check out section 3.4.5 in http://guides.rubyonrails.org/layouts_and_rendering.html#using-partials